torch/fx/_symbolic_trace.py

```diff
@@ -486,8 +489,14 @@ def _module_getattr(self, attr, attr_val, parameter_proxy_cache):
             for n, p in self.root.named_parameters():
                 if attr_val is p:
                     if n not in parameter_proxy_cache:
-                        parameter_proxy_cache[n] = self.create_proxy('get_attr', n, (), {})
+                        if not self.param_shapes_constant:
+                            val_proxy = self.create_proxy('get_attr', n, (), {})
+                        else:
+                            val_proxy = self.create_proxy('fixed_shape_param', n, (), {}, obj_proxied=attr_val)
```

Comment:
We shouldn't use the `kind` field to signal that we want special handling here. The `kind` field is defined as one of the opcodes for `Node` (`call_module`, `call_method`, etc.)

Instead, branch should simply choose which class of `Proxy` we are going to return. One way to do this could be to add an argument `factory_fn` to `create_proxy` that takes a callable that will specify what the call target is here: https://github.com/pytorch/pytorch/blob/26d17ddc9f4f765ac07d550f296b0b35340a3a34/torch/fx/proxy.py#L46


In the default case, it could be `lambda n: self.proxy(n)`, but in the case that `self.param_shapes_constant` is true, it could be something like `lambda n: FixedShapeParam(n, tracer, name, attr_val)`.